src/realm/sort_descriptor.cpp

```diff
@@ -391,6 +403,37 @@ void FilterDescriptor::execute(const Table& table, KeyValues& key_values, const
     key_values = std::move(filtered);
 }
 
+template <typename T, typename U>
+inline int compare(const T& i, const T& j, const U& col)
+{
+    const ColKey col_key{col.col_key};
+    const auto table = col.table;
+    const auto has_interner = col_key.get_type() == col_type_Mixed || col_key.get_type() == col_type_String;
```

Comment:
I think you need to check if the column has a index as well right? Or do you intern strings in a nested collection?

Reply:
We intern also nested collections. At least we should :-)